Fix outdated or missing attributes for travis_ci_mega installation

thelinuxkid opened the pull request:
This stops rvm from prompting for a password when running `apt-get update`

meatballhat commented:
@thelinuxkid Hey, sorry for the delay.  Can you resolve conflicts pretty please? :heart:

thelinuxkid commented:
Done

thelinuxkid commented:
I removed the commit about ````go clean` before installing go packages in travis_build_environment. This avoids an error over conflicting versions of go in the package```
```
  diff --git a/cookbooks/travis_build_environment/recipes/ci_user.rb
  b/cookbooks/travis_build_environment/recipes/ci_user.rb
  index 08423b1..01b8129 100644
  --- a/cookbooks/travis_build_environment/recipes/ci_user.rb
  +++ b/cookbooks/travis_build_environment/recipes/ci_user.rb
  @@ -133,7 +133,7 @@ gimme_versions.each do |version|

     Array(node['travis_build_environment']['golang_libraries']).each do
     |lib|
       bash "install #{lib} for go #{version}" do
  -      code %{eval "$(gimme #{version})" && go get -u #{lib}}
  +      code %{eval "$(gimme #{version})" && go clean -i -r #{lib} && go get
     -u #{lib}}
         flags '-l'
         user node['travis_build_environment']['user']
         group node['travis_build_environment']['group']
```
because it bombs when `go clean` doesn't have anything to clean. I replaced it with a guard:
```
diff --git a/cookbooks/travis_build_environment/recipes/ci_user.rb b/cookbooks/travis_build_environment/recipes/ci_user.rb
index 01b8129..0be87d9 100644
--- a/cookbooks/travis_build_environment/recipes/ci_user.rb
+++ b/cookbooks/travis_build_environment/recipes/ci_user.rb
@@ -132,8 +132,18 @@ gimme_versions.each do |version|
   next if version < '1.4'

   Array(node['travis_build_environment']['golang_libraries']).each do |lib|
+    bash "clean #{lib} for go #{version}" do
+      code %{eval "$(gimme #{version})" && go clean -i -r #{lib}}
+      flags '-l'
+      user node['travis_build_environment']['user']
+      group node['travis_build_environment']['group']
+      environment('HOME' => node['travis_build_environment']['home'])
+      only_if do code %{eval "$(gimme #{version})" && go clean -i -r #{lib}}, environment => {
+        'HOME' => node['travis_build_environment']['home']
+      } end
+    end
     bash "install #{lib} for go #{version}" do
-      code %{eval "$(gimme #{version})" && go clean -i -r #{lib} && go get -u #{lib}}
+      code %{eval "$(gimme #{version})" && go get -u #{lib}}
       flags '-l'
       user node['travis_build_environment']['user']
       group node['travis_build_environment']['group']
```
But that's throwing a syntax error. I tried different variations from https://docs.chef.io/resource_common.html but I couldn't get any of those to work either. Maybe you can help? It needs to run `go clean` only when the go package, i.e., #{lib}, exists.

Is this even the right solution? Should it be installing to $GOROOT, e.g., /home/travis/.gimme/versions/go1.5.1.linux.amd64, instead so that there's no conflict in the package builds?
